lib/concurrent/promise.rb

```diff
@@ -438,6 +438,46 @@ def self.any?(*promises)
       aggregate(:any?, *promises)
     end
 
+    # Aggregates a collection of promises, and completes as soon as the first
+    # aggregated promise completes, with the same state (fulfilled or rejected)
+    # and reason/value.
+    #
+    #   @param [Array] promises Zero or more promises to aggregate
+    #   @return [Promise] an unscheduled (not executed) promise that aggregates
+    #     the promises given as arguments
+    def self.race(*promises)
+      return Promise.fulfilled if promises.empty?
+
+      Promise.new do
+        latch = Concurrent::CountDownLatch.new(1)
+        finished = Concurrent::AtomicBoolean.new(false)
```

Comment:
Maybe it can be done? We need something that we can, atomically, set once and see if our set took (or someone else set first). Can I do this with a latch somehow?  The `make_true` on the atomic boolean is crucial, is there an equivalent with the latch?